crates/krun/src/bin/krun.rs

```diff
@@ -58,10 +60,17 @@ fn main() -> Result<()> {
     };
 
     {
-        // Configure the number of vCPUs (4) and the amount of RAM (4096 MiB).
+        let ram_mib = if let Some(ram_mib) = options.mem {
+            ram_mib
+        } else {
+            let sysinfo = sysinfo().context("Failed to get system information")?;
+            let ram_total = sysinfo.ram_total() / 1024 / 1024;
+            cmp::min(MiB::from((ram_total as f64 * 0.8) as u32), MiB::from(16384))
+        };
+        // Configure the number of vCPUs (4) and the amount of RAM (max 16384 MiB).
```

Comment:
(Moved from https://github.com/teohhanhui/krun/commit/ce29e32992389569b70076ccc791577b8d0cfe28#r142414585)

@asahilina:
> Aside, but for vCPUs, what I would do is enumerate the system CPUs, filter out the P-cores (the cores with the highest capacity), use the count as the vCPU count, and then automatically pin the krun VM to those CPUs. That sidesteps the big.LITTLE VM scheduling issue (which is a much bigger problem) ^^

Reply:
Sounds like a good strategy to me too. @teohhanhui do you want to extend this PR to cover that or should we leave that for later.